core/src/test/java/org/elasticsearch/common/lucene/all/SimpleAllTests.java

```diff
@@ -374,4 +374,13 @@ public void testNoTokensWithKeywordAnalyzer() throws Exception {
         assertThat(docs.totalHits, equalTo(1));
         assertThat(docs.scoreDocs[0].doc, equalTo(0));
     }
+
+    public void testEquals() {
+        Term bar = new Term("foo", "bar");
+        Term baz = new Term("foo", "baz");
+        assertEquals(new AllTermQuery(bar), new AllTermQuery(bar));
+        assertFalse(new AllTermQuery(bar).equals(new AllTermQuery(baz)));
```

Comment:
Maybe rewrite this using `assertNotEquals`? It makes it symmetric with the previous line so that it's clearer, and it gives better failure messages from JUnit when the assertion fails.